spark/spark-spa/src/main/java/com/thoughtworks/go/spark/spa/AnalyticsDelegate.java

```diff
@@ -0,0 +1,145 @@
+
+    private final SPAAuthenticationHelper authenticationHelper;
+    private final TemplateEngine engine;
+    private final SystemEnvironment systemEnvironment;
+    private final AnalyticsExtension analyticsExtension;
+    private final PipelineConfigService pipelineConfigService;
+
+    public AnalyticsDelegate(SPAAuthenticationHelper authenticationHelper, TemplateEngine engine, SystemEnvironment systemEnvironment, AnalyticsExtension analyticsExtension, PipelineConfigService pipelineConfigService) {
+        this.authenticationHelper = authenticationHelper;
+        this.engine = engine;
+        this.systemEnvironment = systemEnvironment;
+        this.analyticsExtension = analyticsExtension;
+        this.pipelineConfigService = pipelineConfigService;
+    }
+
+    @Override
+    public String controllerBasePath() {
+        return Routes.AnalyticsSPA.BASE;
+    }
+
+    @Override
+    public void setupRoutes() {
+        path(controllerBasePath(), () -> {
+            before("", authenticationHelper::checkUserAnd403);
+            get("", this::index, engine);
+        });
+
+        path(controllerPath(SHOW_PATH), () -> {
+            before("", this::checkPipelineExists, this::checkPermissions);
+            before("/*", this::checkPipelineExists, this::checkPermissions);
+            get("", this::showAnalytics);
+        });
+    }
+
+    public ModelAndView index(Request request, Response response) {
+        HashMap<Object, Object> object = new HashMap<Object, Object>() {{
+            List<String> pipelines = new ArrayList<>();
+            pipelineConfigService.viewableGroupsFor(SessionUtils.currentUsername()).forEach(
+                    (PipelineConfigs config) -> config.getPipelines().forEach(
+                            (p) -> pipelines.add(p.getName().toString())
+                    )
+            );
+            put("viewTitle", "Analytics");
+            put("pipelines", GSON.toJson(pipelines));
+        }};
+        return new ModelAndView(object, "analytics/index.vm");
+    }
+
+    public String showAnalytics(Request request, Response response) {
+        try {
+            final AnalyticsData analytics = analyticsExtension.getAnalytics(
+                    request.params(":plugin_id"),
+                    request.params(":type"),
+                    request.params(":id"),
+                    getQueryParams(request));
+
+            response.type("application/json");
+            return GSON.toJson(analytics.toMap());
+        } catch (Exception e) {
+            LOG.error("Encountered error while fetching analytics", e);
+            throw halt(500, format("Error generating analytics from plugin - %s", request.params(":plugin_id")));
+        }
+    }
+
+    private void checkPipelineExists(Request request, Response response) {
+        if (isPipelineRequest(request)) {
+            if (null == pipelineConfigService.getPipelineConfig(request.queryParams("pipeline_name"))) {
```

Comment:
Use `GoConfigService#pipelineConfigByName` — it's optimized for faster lookups, instead of iterating through all pipelines in the config.